addons/hitreactions/functions/fnc_checkWeaponDrop.sqf

```diff
@@ -0,0 +1,32 @@
+#include "..\script_component.hpp"
+/*
+ * Author: KJW
+ * Checks if unit should drop weapon based on projectile hit info
+ *
+ * Arguments:
+ * 0: surfaceType <STRING>
+ * 1: components <ARRAY>
+ * 2: position <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["", [], [1,1,1]] call ACE_hitreactions_fnc_checkWeaponDrop
+ *
+ * Public: No
+ */
+
+params ["_surfaceType", "_components", "_position"];
+
+private _didHitArm = _components findAny GVAR(armComponents) isNotEqualTo -1;
+private _didHitGun = _surfaceType isEqualTo "" && _components isEqualTo [];  // Could be potential for edge cases here however couldn't find any from testing.
+
+if (!_didHitArm && !_didHitGun) exitWith {}; // Exit if it didn't hit an arm and also didn't hit a gun.
```

Comment:
```suggestion
if !(_didHitArm || _didHitGun) exitWith {};
```
I prefer this condition style but others might disagree. Comment can go, condition is self-explanatory

Reply:
Edited for it, imo previous one is more readable but second is likely slightly more performant